packages/react-router/lib/dom/ssr/single-fetch.tsx

```diff
@@ -148,11 +148,22 @@ export function StreamTransfer({
   }
 }
 
-function handleMiddlewareError(error: unknown, routeId: string) {
-  return { [routeId]: { type: "error", result: error } };
+export function getSingleFetchDataStrategy(
+  manifest: AssetsManifest,
+  ssr: boolean,
+  basename: string | undefined,
+  getRouter: () => DataRouter
+): DataStrategyFunction {
+  let dataStrategy = getSingleFetchDataStrategyImpl(
+    manifest,
+    ssr,
+    basename,
+    getRouter
+  );
+  return async (args) => args.unstable_runMiddleware(dataStrategy);
```

Comment:
Use the new `runMiddleware` API to wrap middleware around our existing `dataStrategy` - instead of calling middleware independently for each fork in the `dataStrategy`